### PR TITLE
UPS: port XML processing to Nokogiri

### PR DESCRIPTION
**Pull request by wvanbergen**
Except for some tests that are skipped, remote tests pass as well.

@Shopify/shipping

### CODE REVIEW

File: test/remote/ups_test.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ def test_tracking_with_bad_number
   end
 
   def test_tracking_with_another_number
-    @carrier.find_tracking_info('1Z12345E6692804405')
+    response = @carrier.find_tracking_info('1Z12345E6692804405')
```
Comment (garethson) [inline test/remote/ups_test.rb]: :+1:

File: test/remote/ups_test.rb [hunk 2]
```diff
@@ -38,7 +38,7 @@ def test_us_to_uk
   end
 
   def test_puerto_rico
-    @carrier.find_rates(
+    response = @carrier.find_rates(
```
Comment (garethson) [inline test/remote/ups_test.rb]: :+1:

**Comment by garethson** [comment]
Diff was a bit tricky to follow but everything looks good. :boat:

**Comment by kmcphillips** [comment]
Nothing stands out. :+1:

**Comment by wvanbergen** [comment]
Now also XML parsing using Nokogiri,

**Comment by garethson** [comment]
@wvanbergen One thing to watch for on these conversions is to ensure that the test data we're parsing has the XML namespaces that production API calls do, otherwise Nokogiri chokes on parsing.

**Comment by wvanbergen** [comment]
Looks like it's fine; remote tests are happy.

**Comment by wvanbergen** [comment]
AFAIK only fedex has namespaces, and we actively remove them before parsing the XML using `gsub` :poop:.

**Pull request merged**